UnitsNet/CustomCode/Quantities/TemperatureDelta.extra.cs

```diff
@@ -47,6 +50,21 @@ public partial struct TemperatureDelta
         {
             return specificEntropy * temperatureDelta;
         }
+
+        /// <summary>
+        ///     Subtract two temperatures to get a <see cref="TemperatureDelta"/> in <paramref name="left"/>'s unit, which you can later convert to any temperature unit.
+        ///     This is useful since Celsius, Fahrenheit and Kelvins don't share a common zero point on the scale and normal subtraction
+        ///     would operate on Kelvins and likely give an unexpected result.
+        ///     Example:
+        ///     double deltaCelsius = TemperatureDelta.FromTemperatures(celsius30, celsius20).DegreesCelsius; // 10 C
+        ///     double wrongDeltaCelsius = (celsius30 - celsius20).DegreesCelsius; // 303.15 K - 293.15 K = 10 K = -263.15 degrees Celsius
+        /// </summary>
+        /// <returns>A temperature delta.</returns>
+        public static TemperatureDelta FromTemperatures(Temperature left, Temperature right)
+        {
+            // TemperatureDeltaUnit enum has identical values to TemperatureUnit, so we can simply cast between them
+            return new TemperatureDelta(left.Value - right.As(left.Unit), (TemperatureDeltaUnit)left.Unit);
```

Comment:
A straight enum cast relies on the ordering in the json file. Perhaps a switch would be better?

Reply:
I guess, I figured since they were generated it would be safe. How about instead enum value ToString() and enum parse? Then it will throw if they are not identical.

Reply:
@tmilnthorp Pushed new commit with a new ToDeltaUnit() method.